tool/sample-synthetic-data.ts

```diff
@@ -0,0 +1,511 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Genie
+//
+// Copyright 2021 The Board of Trustees of the Leland Stanford Junior University
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Jake Wu <jmhw0123@gmail.com>
+
+import * as argparse from 'argparse';
+import * as fs from 'fs';
+import * as Tp from 'thingpedia';
+import * as I18n from '../lib/i18n';
+import * as utils from '../lib/utils/misc-utils';
+import { Ast, Type, SchemaRetriever } from 'thingtalk';
+import { ParaphraseExample, generateExamples } from './autoqa/lib/canonical-example-constructor';
```

Comment:
`generateExamples as generateQueryExamples` to distinguish from `generateActionExamples`